packages/block-library/src/site-title/index.php

```diff
@@ -16,7 +16,7 @@
  */
 function render_block_core_site_title( $attributes ) {
 	$site_title = get_bloginfo( 'name' );
-	if ( ! $site_title ) {
+	if ( ! $site_title || trim( $site_title ) === '' ) {
```

Comment:
```suggestion
	if ( ! trim( $site_title ) ) {
```

Suggestion (non-blocking): We could simplify the check.